aten/src/THCUNN/generic/ClassNLLCriterion.cu

```diff
@@ -44,24 +46,25 @@ void THNN_(ClassNLLCriterion_updateOutput)(
 
   if (reduction == at::Reduction::None && n_dims == 2) {
     THCTensor_(resize1d)(state, output, batch_size);
-    if (weights) {
-      weights = THCTensor_(newContiguous)(state, weights);
-    }
-
-    ClassNLLCriterion_updateOutput_no_reduce_kernel<scalar_t>
-      <<<GET_BLOCKS(batch_size), CUDA_NUM_THREADS, 0, THCState_getCurrentStream(state)>>>(
-        batch_size,
-        toDeviceTensor<scalar_t, 2>(state, input),
-        toDeviceTensor<THCIndex_t, 1>(state, target),
-        toDeviceTensor<scalar_t, 1>(state, output),
-        weights ? THCTensor_(data)(state, weights) : NULL,
-        n_classes,
-        ignore_index);
-
-    THCudaCheck(cudaGetLastError());
-
-    if (weights) {
-      THCTensor_(free)(state, weights);
+    if (batch_size != 0) {
```

Comment:
We should add a comment that this is to workaround the CUDA launch parameters not handling 0

Reply:
addressed